### DISCUSSION TITLE
How to use flink engine in an environment with only yarn-session

### DISCUSSION THREAD

**Discussion by xiaowuzi010599** [apache/kyuubi#4445] [Q&A]
my kyuubi version is 1.5.2
I started yarn-session and shut down the local cluster through stop-cluster.sh, but I received this error
![image](https://user-images.githubusercontent.com/79125320/222879875-e73f6476-f11c-4918-ad7d-9fab61260591.png)
If I start the local cluster through start-cluster, the query can be executed normally, but the corresponding query cannot be seen on yarn-session web ui
![image](https://user-images.githubusercontent.com/79125320/222879947-968c11ef-6d2a-48ff-8a9e-d5c918ffcfe2.png)

I want to know how flink engine works with yarn-session

**Comment by xiaowuzi010599** [comment]
I upgraded the kyuubi version to 1.6.1, configured hadoop_conf_dir and flink_hadoop_classpath according to the documentation, and used topspeedwindowing to verify that there was no problem with my yarn-session, but after using beeline to start kyuubi flink engine, I still tried to connect to the local environment, and I can imagine that an error was reported. What should I do here?，my hadoop version is 3.1.3 @pan3793

**Reply by pan3793** [reply]
You should provide the `yarn.applicaiton.id` and `execution.target` in your flink configuration yaml, or `kyuubi-defaults.conf` w/ additional prefix `flink.`, or put them in JDBC url w/ additional prefix `flink.`

https://kyuubi.readthedocs.io/en/master/client/jdbc/kyuubi_jdbc.html
https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/deployment/resource-providers/yarn/#session-mode

**Reply by xiaowuzi010599** [reply]
Must I manually specify that application id is in the configuration file? Can it be optimized to get applicationid from / tmp/.yarn-properties-hadoop in later versions, or are there any other considerations?

**Reply by pan3793** [reply]
The long term goal is supporting Flink Application mode, then Kyuubi can fully manage the Flink app lifecycle, just like Spark cluster mode.

@link3280 is driving this.

**Reply by link3280** [reply]
Yes, we need Flink application mode which manages Flink clusters automatically, but ATM the application mode doesn't support Flink jobs submitted from gateways, see https://issues.apache.org/jira/browse/FLINK-26541. I'm preparing a Flink improvement proposal. 

Before that, we may need some hacking in Kyuubi to unlock the application mode, which I'm working on lately.
